elastic/kibana
# [Guided onboarding] Use Kibana features to grant access

## Summary

Fixes https://github.com/elastic/kibana/issues/149132

This PR adds a Kibana feature for the guided onboarding plugin for better permissions handling. By default `kibana_admin` and `editor` roles are granted access to guided onboarding. The role `viewer` on the other hand doesn't have enough permissions to see or use guided onboarding. For any roles that don't have the correct permissions, guided onboarding is completely disabled, the same as it's disabled on-prem. 
When creating a new role, the feature "Setup guides" can be enabled or disabled. 

### How to test
1. Add `xpack.cloud.id: 'testID'` to `/config/kibana.dev.yml`
1. Start ES with `yarn es snapshot` and Kibana with `yarn start``
2. Login as elastic and create a test user with the role `viewer`
3. Clear everything from your browser's local storage 
4. Login as the test user and check the following
- On the first visit, the "on-prem" welcome message is shown (not the guided onboarding landing page)
- The url `/app/home#/getting_started` is unknown and redirects back to the home page
- There is no button "Setup guides" in the header
- There is no link "Setup guides" in the help menu

### Checklist
